applications/OptimizationApplication/python_scripts/model_part_controllers/connectivity_preserving_model_part_duplication_controller.py

```diff
@@ -0,0 +1,39 @@
+import KratosMultiphysics as Kratos
+from KratosMultiphysics.OptimizationApplication.model_part_controllers.model_part_controller import ModelPartController
+
+def Factory(model: Kratos.Model, parameters: Kratos.Parameters, _) -> ModelPartController:
+    if not parameters.Has("settings"):
+        raise RuntimeError(f"ConnectivityPreservingModelPartDuplicationController instantiation requires a \"settings\" in parameters [ parameters = {parameters}].")
+    return ConnectivityPreservingModelPartDuplicationController(model, parameters["settings"])
+
+class ConnectivityPreservingModelPartDuplicationController(ModelPartController):
```

Comment:
0 / 10 naming :laughing: 

how about `ElementFactoryController` or `ConditionFactoryController` (with the necessary changes to make it only work for elements/conditions)? Or at least something **much** more concise that explains what the controller does, without encoding the entire source code in the class' name.